python/ray/experimental/data/dataset.py

```diff
@@ -1118,7 +1118,7 @@ def __repr__(self) -> str:
         schema = self.schema()
         if schema is None:
             schema_str = "Unknown schema"
-        elif hasattr(schema, "names"):
+        elif schema and not isinstance(schema, type):
```

Comment:
Incorrect check for pyarrow schema type. Check still looks a bit funny to avoid importing pyarrow.lib.Schema here directly.